concepts/microservices/2-software-architect.md

```diff
@@ -0,0 +1,214 @@
+# Microservice Architects
+
+## Outline
+
+- [Role of Architects](#role-of-architects)
+- [Decisionmaking](#decisionmaking)
+  - [Strategic Goals](#strategic-goals)
+  - [Principles](#principles)
+  - [Practices](#practices)
+- ["Good Citizen" Service](#good-citizen-service)
+  - [Monitoring](#monitoring)
+  - [Interface](#interface)
+  - [Safety](#safety)
+- [Ways to Share Knowledge](#ways-to-share-knowledge)
+  - [Documentation](#documentation)
+  - [Exemplars](#exemplars)
+  - [Service Templates](#service-templates)
+- [Responsibilities](#responsibilities)
+
+## Role of Architects
+
+Architects are in charge of making sure that an organization
+has a technical vision that lines up with the systems that the organization's
+customers need. In short, they are responsible for developing a framework
+in which the right systems can emerge and continue to evolve over time.
+Architects, to borrow an example from Sam Newman, are like town planners
+in that they define architectural direction in broad strokes, and only
+get involved in defining implementation details in specific cases.
+
+Instead of thinking about what happens inside a microservice, architects
+tend to think more about what happens _in between_ microservices. Choices
+made in between microservices tends to be what leads to a messy service
+implementation. What happens when your service leverages multiple
+communication protocols (HTTP, SOAP, gRPC, etc)? Once architects determine
+an optimal course of action, then it's up to them to ensure that the
+trade-offs are worth it.
+
+Here's a good guideline to follow:
+
+> Be worried about what happens between the boxes, and be liberal about
+> what happens inside.
+
+## Decisionmaking
+
+Making decisions as an architect is all about trade-offs:
+
+1. Is it ok for us to have multiple technology stacks in our system?
+1. Do we pick a datastore that we have less experience with, but scales well?
+
+When making decisions, it's important to use a decision framework to determine
+the best course of action:
+
+- Strategic Goals
+- Principles
+- Practices
+
+### Strategic Goals
+
+These are high level goals that are defined at an organizational or group level.
+They are meant to be directional goals that communicate where your organization
+or group is headed, and are more aligned around strategy than anything else. It's
+up to the architect to come up with principles that adhere to these streategic
+goals.
+
+### Principles
+
+Principles are rules that are made in order to align with your strategic goals.
+Generally, principles tend to be constraints that are necessary to reach some
+necessary outcome. For example, Heroku's 12 Factors are mostly a list of constraints
```

Comment:
> Heroku's 12 Factors

A link for this?

Reply:
https://github.com/irmiller22/tamagotchi/pull/1/commits/1384c0d33d52038a9b9887bb900219e921ec6956